heron/api/src/java/com/twitter/heron/api/windowing/triggers/TimeTriggerPolicy.java

```diff
@@ -47,88 +37,45 @@ public TimeTriggerPolicy(long millis, TriggerHandler handler, EvictionPolicy<T>
       evictionPolicy) {
     this.duration = millis;
     this.handler = handler;
-    this.executor = Executors.newSingleThreadScheduledExecutor();
     this.evictionPolicy = evictionPolicy;
   }
 
   @Override
   public void track(Event<T> event) {
-    checkFailures();
+    if (started && event.isTimer()) {
+      triggerTask();
+    }
   }
 
   @Override
   public void reset() {
-    checkFailures();
+
```

Comment:
why the `reset()` doesn't triggerTask here?

Reply:
The point of reset is to clear some internal state of the trigger.  Please look at the CountTriggerPolicy.  The TimeTriggerPolicy does not have any internally held state to clear since its just a timer